File: DesktopCore/src/bean/NumberField.java
package bean;

import javax.swing.JFormattedTextField;
import javax.swing.text.NumberFormatter;

/**
 * Text field to contain integer values
 */
public class NumberField extends JFormattedTextField
{
	public NumberField()
	{
		this(null, null);
	}
	public NumberField(Integer min)
	{
		this(min, null);
	}
	public NumberField(Integer min, Integer max)
	{
		super(new NumberFormatter());
		
		NumberFormatter nf = (NumberFormatter)getFormatter();
		
		if (min != null)
		{
			nf.setMinimum(min);
		}
		
		if (max != null)
		{
			nf.setMaximum(max);
		}
	}
	
	public int getNumber()
	{
		Object val = getValue();
		if (val == null)
		{
			return -1;
		}
		
		return (int)val;
	}
	
	public void setMaximum(int max)
	{
		NumberFormatter nf = (NumberFormatter)getFormatter();
		nf.setMaximum(max);
	}
	public void setMinimum(int min)
	{
		NumberFormatter nf = (NumberFormatter)getFormatter();
		nf.setMinimum(min);
	}
}

File: src/main/kotlin/dartzee/core/bean/NumberField.kt
package dartzee.core.bean

import javax.swing.JFormattedTextField
import javax.swing.text.NumberFormatter

/** Text field to contain integer values */
class NumberField constructor(min: Int? = null, max: Int? = null, val testId: String = "") :
    JFormattedTextField(NumberFormatter()) {
    init {
        if (min != null) {
            setMinimum(min)
        }

        if (max != null) {
            setMaximum(max)
        }
    }

    fun getDouble(): Double {
        val n = value ?: return 0.0

        return if (n is Double) n else (n as Int).toDouble()
    }

    fun getNumber(): Int {
        val n = value ?: return -1
        return n as Int
    }

    fun setMaximum(max: Int) {
        val nf = formatter as NumberFormatter
        nf.maximum = max
    }

    fun setMinimum(min: Int) {
        val nf = formatter as NumberFormatter
        nf.minimum = min
    }

    fun getMaximum() = (formatter as NumberFormatter).maximum
}
